PLUGIN_STYLEGUIDE.md

```diff
@@ -0,0 +1,146 @@
+
+
+    `sensuctl asset add CHANGEME/sensu-CHANGEME:VERSION`
+
+
+    If you're using an earlier version of sensuctl, you can find the asset on the [Bonsai Asset Index]([https://bonsai.sensu.io/assets/CHANGEME/sensu-CHANGEME](https://bonsai.sensu.io/assets/CHANGEME/sensu-CHANGEME)).
+
+In general, avoid recommending building the plugin from source. Instead, use sensuctl asset add or create a definition manually via sensuctl create -f /path/to/asset-definition.yml.
+
+
+### Asset definition
+
+Users can install asset definitions using `sensuctl asset add`, but we recommend including an example asset definition in your plugin README. Format your example in yaml.
+
+
+### Resource definitions
+
+In the resource definitions, include example check, filter, mutator, or handler definitions (depending on the resources in your plugin). Format your resource definitions in yaml.
+
+
+## Installation from source
+
+Assets are the preferred plugin installation method. If you cannot use an asset to install your plugin, make sure to provide all the documentation users will need to compile your plugin from source. Some useful examples of plugins providing instructions for installing from source are:
+
+*   [InfluxDB Handler](https://github.com/sensu/sensu-influxdb-handler#dependencies)
+*   [Slack Handler](https://github.com/sensu/sensu-slack-handler#installing-from-source-and-contributing)
+
+## Additional notes
+
+Use the **Additional notes** section to provide further context for users who will download and deploy your plugin. For example, you might explain whether your plugin requires an on-disk configuration file to properly operate (such as templates for notification plugins) or recommend that users create RBAC roles that allow your plugin to make calls to the Sensu backend API. Describe any additional configuration needed to operate your plugin in this section.
+
+## Contributing
+
+We recommend that all community plugins include a **CONTRIBUTING** document in their repositories. If you create a plugin from the [Sensu Go Plugin](https://github.com/sensu/sensu-go-plugin) repository, a CONTRIBUTING document is included in the repository. If you create a plugin using a different method, you can add your own CONTRIBUTING document. Examples of CONTRIBUTING documents from the community include:
+
+*   [Contributing to Sensu Go](https://github.com/sensu/sensu-go/blob/master/CONTRIBUTING.md)
+*   [Sensu Plugins Developer Guidelines](http://sensu-plugins.io/docs/developer_guidelines.html)
+
+
+# Plugin collections
+
+We recommend adopting the [Unix Philosophy](https://homepage.cs.uri.edu/~thenry/resources/unix_art/ch01s06.html): write a plugin that does one thing well, and avoid collections of plugins. Collections may simplify deployment, but they complicate documenting your plugin’s functionality. 
```

Comment:
While I think this is fair I think it really depends since for example with aws plugins if they were all separate we would likely have each check re-implementing auth in inconsistent ways. This also would mean a lot more maintenance for maintaining say the thousands of community plugins.

Reply:
```suggestion
```

I think this prescription is beyond the scope of this document. Perhaps we can omit it and focus instead on how we think documentation for plugin collections should be organized?

I like how the plugins in https://github.com/ncr-devops-platform/nagiosfoundation each have their own readme, but that organization may not make sense everywhere.

Reply:
@majormoses 
Hey it looks like the guide was updated with a specific recommendation on how to document collections.  Is there anything else outstanding? Or are we good to approve and merge?